### ISSUE TITLE
Tags dissapeared from collections

### ISSUE THREAD

**Issue by difrkaguilar** [darktable-org/darktable#9728]
**Describe the bug/issue**
I had a lot of tagged images, but now in collections non tag appear, all tags of my images disappeared.

**To Reproduce**
1. Go to Collections
2. Click on tag
3. Check available tags
4. In my case non tags appear

**Expected behavior**
Every tag previously created must appear in this section.

**Screenshots**

This is my tags collection before the issue.

![tags_01](https://user-images.githubusercontent.com/67672868/128203752-2376cddb-df20-4b57-bd90-0f193ad16a25.png)

This is my tags collection at this moment.

![Screenshot-20210804113029-326x290](https://user-images.githubusercontent.com/67672868/128209536-e7d29fb6-2611-4900-9539-7f7c1eabc7d9.png)

**Which commit introduced the error**
3.7.0~git648.cbadef13f2-6384.1 or before. I'm not sure.

**Platform**
darktable 3.7.0~git648.cbadef13f2-6384.1
bit depth is 64 bit
normal build
SSE2 optimized codepath enabled
OpenMP support enabled
OpenCL support enabled
Lua support enabled, API version 8.0.0-dev
Colord support enabled
gPhoto2 support enabled
GraphicsMagick support enabled
ImageMagick support disabled
OpenEXR support enabled

OS: Fedora 34 (Workstation Edition) x86_64
Host: Z390 AORUS MASTER
Kernel: 5.12.17-300.fc34.x86_64
Uptime: 1 hour, 14 mins
Packages: 3107 (rpm), 8 (flatpak), 12 (snap) 
Shell: zsh 5.8
Resolution: 3840x2160
DE: GNOME 40.3
WM: Mutter
WM Theme: Materia-dark-compact
Theme: Mojave-dark [GTK2/3]
Icons: Os-Catalina-icons-master [GTK2/3]
Terminal: gnome-terminal
CPU: Intel i7-9700K (8) @ 4.900GHz
GPU: NVIDIA GeForce GTX 1070
GPU: NVIDIA GeForce GTX 1080 Ti
Memory: 2219MiB / 32037MiB
Graphics driver : 460.73.01
OpenCL installed : yes
OpenCL activated : yes

**Comment by Nilvus** [comment]
I can't reproduce nor in 3.7.0git643 nor with 3.7.0git649 and regarding what commits between them, I'm quite sure there no related issue in commits between. I suppose that you compile darktable; Did you correctly remove build folder inside source code one and darktable install folder before compiling? Could you try again?

**Comment by difrkaguilar** [comment]
I installed from here:

dnf config-manager --add-repo https://download.opensuse.org/repositories/graphics:darktable:master/Fedora_Rawhide/graphics:darktable:master.repo
dnf install darktable

I'll check once again to see if it is my mistake.

**Comment by phweyland** [comment]
Could you share the console messages file (darktable -d all) ?
Can you check the tags are still attached to the images ?

**Comment by difrkaguilar** [comment]
> Could you share the console messages file (darktable -d all) ?

I leave the log to your consideration.
[darktableLog.txt](https://github.com/darktable-org/darktable/files/6948538/darktableLog.txt)

> Can you check the tags are still attached to the images ?
I have checked the images tagged as: isis_flores(55) must be 55 images with this tag but there are none. Days before I uploaded an screenshot to another issue, the #9695 and the 55 images of the model was tagged ok. I left you the screenshot as reference.

![Screenshot-20210731120800-2896x1860](https://user-images.githubusercontent.com/67672868/128586601-f3df73cc-4a1d-4ab4-9900-3cb8838ca1a4.png)

**Comment by johnny-bit** [comment]
open up `data.db` in sqlite3 and do `select * from tags;` - should show all tags that are in db.

**Comment by phweyland** [comment]
> > Could you share the console messages file (darktable -d all) ?
> 
> I leave the log to your consideration.
> [darktableLog.txt](https://github.com/darktable-org/darktable/files/6948538/darktableLog.txt)

Sorry, I should have given more instructions. I want to have the trace of the collect module when set on "tags" and doesn't provide the right collections.
In your trace I've only got the film_roll queries (you haven't requested the tags collections).



> I left you the screenshot as reference.

This is the case which works. With dt which doesn't work (tags collections missing), you select an image, open the tagging the module, and show what it's displayed there.

**Comment by difrkaguilar** [comment]
Finally I make a not great solution. I renamed the data.db to data_Old.db and use the data.db-pre-3.7.0 because this db have all tags of my collection. I made a copy of data.db-pre-3.7.0 and renamed to data.db

The original data.db  was created: Thu 29 Apr 2021 06∶21∶55 PM CDT
![Screenshot-20210807115956-526x565](https://user-images.githubusercontent.com/67672868/128606499-0b8560f2-12bd-4390-9b29-16b6adf6a11d.png)
I made a copy of data.db-pre-3.7.0 and renamed to data.db

![Screenshot-20210807113957-1530x837](https://user-images.githubusercontent.com/67672868/128606518-a7c87e38-4fd2-496b-9278-0c80e9e1b201.png)

I know this is not the real solution but I don't know what's really happen.
Now I have my tags back.
![Screenshot-20210807121417-3840x2160](https://user-images.githubusercontent.com/67672868/128606652-b59a41b1-fe9b-4e35-80f2-c34efe9cc216.png)
